sys/netlink/netlink_message_parser.h

```diff
@@ -47,11 +50,21 @@ static inline void *
 lb_alloc(struct linear_buffer *lb, int len)
 {
 	len = roundup2(len, _Alignof(__max_align_t));
+#ifdef __CHERI_PURE_CAPABILITY__
+	len = CHERI_REPRESENTABLE_LENGTH(len);
+	char *data = __align_up(lb->base + lb->offset,
+	    CHERI_REPRESENTABLE_ALIGNMENT(len));
+	if (data + len > lb->base + lb->size)
+		return (NULL);
+	lb->offset = (data + len) - lb->base;
+	return (cheri_setboundsexact(data, len));
+#else
 	if (lb->offset + len > lb->size)
 		return (NULL);
 	void *data = (void *)(lb->base + lb->offset);
 	lb->offset += len;
 	return (data);
+#endif
```

Comment:
So something like this?
```suggestion
	len = CHERI_REPRESENTABLE_LENGTH(len);
	char *data = __align_up(lb->base + lb->offset,
	    CHERI_REPRESENTABLE_ALIGNMENT(len));
	if (data + len > lb->base + lb->size)
		return (NULL);
	lb->offset = (data + len) - lb->base;
#ifdef __CHERI_PURE_CAPABILITY__
	return (cheri_setboundsexact(data, len));
#else
	return (data);
#endif
```

Reply:
Bleh, this gives errors on non-CHERI:

```
/local/scratch/jenkins/workspace/CheriBSD-pipeline_PR-1826/cheribsd-aarch64-build/local/scratch/jenkins/workspace/CheriBSD-pipeline_PR-1826/cheribsd/arm64.aarch64/tmp/usr/include/netlink/netlink_snl.h:107:6: error: aligning a value to 1 byte is a no-op [-Werror,-Wtautological-compare]
            CHERI_REPRESENTABLE_ALIGNMENT(len));
            ^
```

Reply:
:( that is unfortunate and I can't think of a way of silencing that warning...

Reply:
Unless we add a new macro like CHERI_ALIGN_TO_REPRESENTABLE that is just a noop for non-CHERI?

Reply:
In the latest version I was able to share a bit more of the code, but at the cost of more #ifdef's

Reply:
A wrapper macro around the actual alignment might be nice though (and probably a bit more readable).  We should see if any other places would be simplified by such a macro.

Reply:
Oh, I guess there is CHERI_REPRESENTABLE_BASE that I can use already?

Reply:
Ah, that is a rounddown and I need a roundup

Reply:
See #1832